pyogrio/_io.pyx

```diff
@@ -1791,6 +1823,8 @@ cdef create_ogr_dataset_layer(
         if driver == 'ESRI Shapefile':
             # Fiona only sets encoding for shapefiles; other drivers do not support
             # encoding as an option.
+            if encoding is None:
```

Comment:
I'm not quite sure about the order of operations and the interactions around setting the `"ENCODING"` option for writing datasets.

I think the previous implementation was setting this to blank (which then I think means refer to the `.cpg` file), and using `UTF-8` for shapefiles if supported, falling back to `ISO-8859-1` when `UTF-8` is not supported.

Now, this will always use `ISO-8859-1` regardless of whether the capability for `UTF-8` exists.

This came up in for reading [#138](https://github.com/geopandas/pyogrio/issues/138#issuecomment-1179288076) and we fixed the order of operations for reading in #139.

Reply:
In the previous implementation, right before calling create_ogr_dataset_layer this was done to the encoding:
``` py
    if not encoding:
        encoding = locale.getpreferredencoding()
```
So, in effect, when creating a shapefile, if the encoding wasn't specified, the file would be created as UTF-8 on linux/MacOS and in cp1552 for windows (the preferred encodings of the platform.

UTF-8 being supported or not isn't really a thing when creating a new file. There doesn't exist a `.cpg` file yet, do it can also not be checked whether it contains 'UTF-8' or not. So, the `.cpg` and the "UTF-8" capability is only relevant for reading.